core/src/main/java/org/mule/runtime/core/api/policy/PolicyChain.java

```diff
@@ -53,7 +83,31 @@ public void setProcessors(List<Processor> processors) {
 
   @Override
   public final void initialise() throws InitialisationException {
-    processorChain = new DefaultMessageProcessorChainBuilder().chain(this.processors).build();
+    DefaultMessageProcessorChainBuilder chainBuilder = new DefaultMessageProcessorChainBuilder().chain(processors);
+
+    List<Processor> beforeNext = new ArrayList<>();
+    List<Processor> afterNext = new ArrayList<>();
+
+    boolean seenNext = false;
+    for (Processor processor : processors) {
+      if (processor instanceof PolicyNextActionMessageProcessor) {
+        seenNext = true;
+        continue;
+      }
+
+      if (seenNext) {
+        afterNext.add(processor);
+      } else {
+        beforeNext.add(processor);
+      }
+    }
+
+    processingStrategy = new SourcePolicyProcessingStrategy(schedulerService, muleContext.getSchedulerBaseConfig(),
+                                                            getLocation().getLocation(), beforeNext, afterNext);
+
+    chainBuilder.setProcessingStrategy(processingStrategy);
+    processorChain = chainBuilder.build();
+    chainWithPs = processingStrategy.onPipeline(processorChain);
```

Comment:
I'm wondering if it might be cleaner for processing strategy to have `onChain` that can optionally be implemented and would be used one for each chain.  I'm not that convinced though because in general chains should be transparent and it shouldn't really matter, or change behaviour, if 8 processors are in one chain or two.  So maybe this is simply a special case, that while not that elegant is best we handle like this.  Just thoughts..